app/models/concerns/test_running.rb

```diff
@@ -7,7 +7,7 @@ def run_tests!
 
   def new_test_server_request
     {test: exercise.test,
-     extra: exercise.extra_code || '',
+     extra: exercise.get_extra_code,
```

Comment:
I'd try to find a better name `get_extra_code` sounds a little odd, and doesn't express that it actually represents the whole extra_code, instead of the exercise's  own extra_code.

What about `full_extra_code`, or something like that? Or perhaps overriding the extra_code writer method, so that it answers the concatenation of both attributes? [Example](http://stackoverflow.com/questions/21835116/overwrite-getter-activerecord-model-rails)